tunnel-server/src/proxy.ts

```diff
@@ -96,9 +127,11 @@ export function proxyHandlers({
       }
 
       if(env.access === 'private'){
-        // need to support session cookie, native browser Websocket api doesn't forward authorization header
-        socket.end();
-        return;
+        const session = sessionManager(req, undefined as any, env.clientId)
+        if (session.user?.role !== 'admin'){
+          socket.end();
```

Comment:
```suggestion
          socket.end('Invalid role');
```

Reply:
It can be 'forbidden' or 'unauthorized', but I'm not sure how the client can consume it.